### ISSUE TITLE
Error 1 displayed when switching to an empty window on 16C.

### ISSUE THREAD

**Issue by meltdown03** [zathras/jrpn#109]
**Is your feature request related to a problem? Please describe.**

The Error 1 should only be displayed when switching to a window not through 0 to 7, not just if the window is empty.

**Describe the solution you'd like**

`f WINDOW` should goto that window regardless of the contents. 

**Additional context**

Page 33 of the manual states that these windows will be blank:
"(With smaller word sizes or
smaller numbers, the higher windows will be blank.) **Error 1** results
if you specify a window number greater than 7."

**Other things**
I noticed when switching between bases, the window stays on the same screen if there enough digits in both bases to have the same window being shown.
Ex:
`0 f WSIZE` (64 bit mode)
`g SF 3`
`BIN`
`100000000 ENTER` -- 00000000 .b
`f WINDOW 1` -- 000000001 .b.
`HEX` -- 00000000 h. (still at window 1)

If you preform all the steps above *except* `g SF 3` it will show: `100 h` as soon as you switch to HEX.

**Comment by zathras** [comment]
Do you have a real 16C?  If so...

What does "blank" mean, especially when leading zeroes aren't shown?  Is it really blank, like no digits displayed (maybe just .h?)  Or does it put a zero in there?

About base switching, what does a real 16C do when those keystrokes are entered?

**Comment by meltdown03** [comment]
Yes, I have one, "blank as in no digits, just the `h.` or whatever mode. The dot after the mode, indicating more digits to the right of the current window.
As for switching bases when on any window, it always goes to window 0 after switching.

**Comment by meltdown03** [comment]
I've also just noticed JRPN has the "." before the mode letter when leading zeros are ON, but it's only zeros in all the windows higher than the current one. 
The actual 16C only shows the dot if there is a non-zero in any higher windows.

such as:
`g SF 3`
`HEX 100`
`ENTER` 
 DISPLAY: `00000100 .h`

On the actual device:
`00000100 h`

**Comment by zathras** [comment]
@meltdown03 What happens when leading zeros are being shown (flag 3 set)?  Say, I'm in hex mode, and I do f WINDOW 7.  Do you see "00000000 h." in this case?

**Comment by zathras** [comment]
@meltdown03 Another question:  How does ">" behave?

Say, for example, I'm not showing leading zeroes, 0x123 is displayed, and I press "> > > > > < < < < <".  What do I see?  I put question marks where it's not clear:

```
  123 h
>  12 h.
>   1 h.
> ??? h.
> ??? h.
> ??? h.
< ??? h.
< ??? h.
< ??? h?
< ??? h?
< 123 h
```

**Comment by meltdown03** [comment]
I'll check in a couple hours, what word size do you want me to test it in?

**Comment by zathras** [comment]
Oh, interesting.  I was thinking 64, but maybe also 32?  Thanks!

**Comment by meltdown03** [comment]
> @meltdown03 What happens when leading zeros are being shown (flag 3 set)? Say, I'm in hex mode, and I do f WINDOW 7. Do you see "00000000 h." in this case?

In this case (2-32-1000), I just get a blank display where digits would be and `h.` for the mode.

**Comment by meltdown03** [comment]
> @meltdown03 Another question: How does ">" behave?
> 
> Say, for example, I'm not showing leading zeroes, 0x123 is displayed, and I press "> > > > > < < < < <". What do I see? I put question marks where it's not clear:
> 
```
  123 h      (correct)
>  12 h.     (correct)
>   1 h.     (correct)
> ??? h.     (no digits) h.
> ??? h.     (no digits) h.
> ??? h.     (no digits) h.
< ??? h.     (no digits) h.
< ??? h.     (no digits) h.
< ??? h?               1 h.
< ??? h?              12 h.
< 123 h      (correct)
```

**Comment by zathras** [comment]
Interesting, thanks.  So it looks like with leading zeros, it only left-pads out to the word size.  For windowing, one essentially scrolls across space for 64 digits, regardless of what base it's.  This behavior definitely wouldn't have been my first guess, but it's easy enough to implement!

**Comment by zathras** [comment]
@meltdown03 One last question, to double-check:  What happens to the leading dot when leading zeros are scrolled off to the left?  That is to say, if you're in hex mode, 64 bit, and you enter:

0 ENTER f  WINDOW 0

Do you see this?
```
00000000 .h
```
or does it suppress the leading "." because they're just leading zeroes?

**Comment by meltdown03** [comment]
It does have the leading dot.

**Comment by zathras** [comment]
Thanks!

**Comment by meltdown03** [comment]
I noticed the latest commits and the "Fixed in repo" label. I gave it a try on the Desktop and Android versions, but am unable to scroll to the right at all now doing that test (2-64-0000, no leading zeros): HEX 123 > > >. Also no matter what window I select, I always see: `123 h' Maybe I just jumped the gun on test driving it.

**Comment by zathras** [comment]
@meltdown03 Oh, I see my mistake.  Could you try it out now?  Thanks.

**Comment by meltdown03** [comment]
Yeah, it's working now. Is doesn't set the ERROR 1 when trying to goto window 8 or higher though.

**Comment by zathras** [comment]
@meltdown03  What does the real calculator do when you're in program entry mode and you press f-WINDOW-8 or f-WINDOW-9?  How about f-WINDOW-C?

**Comment by meltdown03** [comment]
Error 1 for all 3, in program mode or run mode

**Comment by zathras** [comment]
@meltdown03 Interesting.  Do SF and CF behave the same way?  For example, when not entering a program, does g-CF-9 and g-CF-A result in a digit entry, or an Error 1?

**Comment by meltdown03** [comment]
Yes it does, Error 1 for those also.

**Comment by meltdown03** [comment]
Same, Error 1

**Comment by zathras** [comment]
OK, thanks.  That's a little unexpected, in that it's different from how other erroneous keystrokes are handled.  It doesn't come up on the 15C - all the keystroke sequences that take an argument like this accept 0..9 or 0..19 (via dot).

**Comment by zathras** [comment]
Released in 2.1.15
